src/EFCore.MySql/Update/Internal/MySqlUpdateSqlGenerator.cs

```diff
@@ -3,132 +3,473 @@
+                modificationCommands,
+                writeOperations,
+                PositionColumnName);
+            AppendOutputClause(
+                commandStringBuilder,
+                keyOperations,
+                InsertedTableBaseName,
+                commandPosition,
+                FullPositionColumnName);
+            commandStringBuilder.AppendLine(SqlGenerationHelper.StatementTerminator);
+
+            AppendSelectCommand(commandStringBuilder, readOperations, keyOperations, InsertedTableBaseName, commandPosition, name, schema, orderColumn: PositionColumnName);
+
+            return ResultSetMapping.NotLastInResultSet;
+        }
+
+        private ResultSetMapping AppendBulkInsertWithServerValuesOnly(
+            StringBuilder commandStringBuilder,
+            IReadOnlyList<ModificationCommand> modificationCommands,
+            int commandPosition,
+            List<ColumnModification> nonIdentityOperations,
+            List<ColumnModification> keyOperations,
+            List<ColumnModification> readOperations)
+        {
+            AppendDeclareTable(commandStringBuilder, InsertedTableBaseName, commandPosition, keyOperations);
+
+            var name = modificationCommands[0].TableName;
+            var schema = modificationCommands[0].Schema;
+            AppendInsertCommandHeader(commandStringBuilder, name, schema, nonIdentityOperations);
+            AppendOutputClause(commandStringBuilder, keyOperations, InsertedTableBaseName, commandPosition);
+            AppendValuesHeader(commandStringBuilder, nonIdentityOperations);
+            AppendValues(commandStringBuilder, nonIdentityOperations);
+            for (var i = 1; i < modificationCommands.Count; i++)
+            {
+                commandStringBuilder.Append(",").AppendLine();
+                AppendValues(commandStringBuilder, nonIdentityOperations);
             }
+            commandStringBuilder.Append(SqlGenerationHelper.StatementTerminator);
+
+            AppendSelectCommand(commandStringBuilder, readOperations, keyOperations, InsertedTableBaseName, commandPosition, name, schema);
 
-            return resultSetCreated ?
-                defaultValuesOnly
-                    ? ResultSetMapping.LastInResultSet
-                    : ResultSetMapping.NotLastInResultSet
-                : ResultSetMapping.NoResultSet;
+            return ResultSetMapping.NotLastInResultSet;
         }
 
-        // ReSharper disable once ParameterTypeCanBeEnumerable.Local
-        private void AppendInsertOutputClause(
+        private void AppendMergeCommandHeader(
+            [NotNull] StringBuilder commandStringBuilder,
+            [NotNull] string name,
+            [CanBeNull] string schema,
+            [NotNull] string toInsertTableAlias,
+            [NotNull] IReadOnlyList<ModificationCommand> modificationCommands,
+            [NotNull] IReadOnlyList<ColumnModification> writeOperations,
+            string additionalColumns = null)
+        {
```

Comment:
This is the function that needs to be re-worked for MySQL Batch Inserts to work.  I can't tell what SQL Server is doing here so I need to dive deeper into this.